### PR TITLE
Refactor discipline saving

### PR DESCRIPTION
**Pull request by TheCarnalest**
## About The Pull Request
Refactors the way Disciplines are read, written, and stored in the backend to use a list instead of preset variables. The Disciplines that are currently in use by a mob are now stored in that mob's Kindred species, since uh, that's just the place where vampire stuff has to go if it needs to be vampire-specific. I don't like it but whatever.

As part of Disciplines no longer being hardcapped to 4, I've implemented some mechanics for learning an unlimited amount of Disciplines. This PR lets others feed you their blood, then teach you the Disciplines they know by spending 10xp. Clan-specific Disciplines can be learned this way, but it alerts the mentor that they are giving away VALUABLE SECRETS.

True Brujah or Old Clan Tzimisce trying to learn Celerity and Vicissitude will find that their bloodline has kicked them out for trying to be like the mainstream Clan. Caitiffs, due to having no Clan, are able to learn any common Discipline without needing to find a tutor first. They however get an experience cost debuff to all Disciplines, needing to spend 6x current rank to upgrade them.

The cost to buy the 1st rank of a Discipline (for the student) is 10 experience points. The cost to upgrade in-Clan Disciplines stays 5x current rank, but the cost for upgrading out-of-Clan Disciplines is 7x current rank as per tabletop. This PR messes with a lot of backend code so that Disciplines don't give you keys for the Clan's hideout for instance.

## Why It's Good For The Game
general code cleanliness and FUCKING USABILITY, along with the roleplay potential of letting people teach each other Disciplines

## Changelog
:cl:
refactor: Disciplines have been completely refactored into a list of Discipline objects stored on vampire players' species datum. 
tweak: Disciplines have been uncapped. You can theoretically have every Discipline.
add: You can now teach your Disciplines to other players. Use the new verb in the IC tab to do so to players you've bloodbonded!
add: Caitiffs can now buy all common Disciplines without needing a mentor. Please don't do evil with this.
/:cl:

### CODE REVIEW

**Comment by wod13-server** [comment]
#### Test Merge Deployed

<details><summary>Details</summary>

##### Server Instance
Test Server

##### Merged By
Lucia

##### Revision
Origin: 620d1a06429c2979ccdce9abb93e6f0806714ec3
Pull Request: 20ac9bdc00ddb599372b52729a1983fc3264b5eb
Server: e63cb69bd7b1226f97323ad06572673b8a7ade58

##### Comment

</details>

**Comment by wod13-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Test Server

##### Merged By
Lucia

##### Revision
Origin: 620d1a06429c2979ccdce9abb93e6f0806714ec3
Pull Request: 871e68004d2111917ffa20622989e240bcd87463
Server: cff9eccfa6aa5e32ed4d40a99c87b8df2b7f3829

##### Comment

</details>

**Comment by wod13-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Test Server

##### Merged By
Lucia

##### Revision
Origin: 620d1a06429c2979ccdce9abb93e6f0806714ec3
Pull Request: e1e50eeeb5b0a873c41a40dd265a115a5ecd88ac
Server: f2a72fc329dd431d3da1d851a315aaa85d3ed206

##### Comment

</details>

**Review by lucifernix — APPROVED** [review]
I think this looks good. I looked over pretty much all of it and didn't spot anything wrong, but I am not the greatest at this.

File: code/modules/vtmb/vampire_clane/true_brujah.dm [hunk 1]
```diff
@@ -47,6 +47,10 @@
 /proc/tempstep(mob/living/M)
 	if(M.temporis_visual)
 		return
+	var/mob/living/carbon/carbon = M
+	if (carbon.celerity_visual)
+		carbon.gib()
+		return
```
Comment (Agateophobia) [inline code/modules/vtmb/vampire_clane/true_brujah.dm:53]: Gibbing is super harsh imho, especially without a warning or anything.

File: code/modules/vtmb/vampire_clane/true_brujah.dm [hunk 2]
```diff
@@ -85,6 +89,10 @@
 /proc/clothogift(mob/living/M)
 	if(M.temporis_blur)
 		return
+	var/mob/living/carbon/carbon = M
+	if (carbon.celerity_visual)
+		carbon.gib()
+		return
```
Comment (Agateophobia) [inline code/modules/vtmb/vampire_clane/true_brujah.dm:95]: But if you have to gib them, at least give it a cool blurb that gets sent to chat after that nonsense happens so people know what  the fuck happened!

**Review by Agateophobia — CHANGES_REQUESTED** [review]
Besides rather harsh punishment for mixing two disciplines, this seems fine from my cursory review, obviously some testing is required to make sure something doesn't explode, but I'm not sure this is exactly testmergeable without causing funny business.

File: code/modules/vtmb/vampire_clane/true_brujah.dm (hunk 1 above)
Comment (lucifernix) [reply code/modules/vtmb/vampire_clane/true_brujah.dm:53]: I thought the same looking at this, lucia is pretty gib-happy lately!

**Review by Agateophobia — APPROVED** [review]

**Comment by wod13-server** [comment]
#### Test Merge Removed

**Pull request merged**